src/CookieActionComponent.cpp

```diff
@@ -161,21 +162,44 @@ void CookieActionComponent::checkAndPerformAction(double deltaTime, double total
             gameManager.getGameWorld().addDynamicGameObject(cookieObj);
 
             previousCookieTime = totalTime;
+            aimInputComponent->pressTime = 0.0;
+
+            // Reset the material of the aiming objects
+            gameObj->changeMaterial(materialManager.getMaterial("Brass"));
+            gameObj1->changeMaterial(materialManager.getMaterial("Brass"));
+            gameObj2->changeMaterial(materialManager.getMaterial("Brass"));
+            gameObj3->changeMaterial(materialManager.getMaterial("Brass"));
+            gameObj4->changeMaterial(materialManager.getMaterial("Brass"));
         }
     }
 
+    glm::vec3 aimTarget2 = holder_->getPosition() + throwDirection * 0.5f;
+    gameObj2->setPosition(aimTarget2);
+    if (chargeTime > 0.01 && chargeTime <= 0.3) {
+        gameObj2->changeMaterial(materialManager.getMaterial("Emerald"));
+    }
```

Comment:
Idk actually, @nurgan would know more since it's his code I moved around. So we can see what he has to say.